### PR TITLE
SurfaceMagnetism behavior

### PR DESCRIPTION
**Pull request by CraigFeldspar**
This PR adds a behavior for meshes to stick on the surfaces of other meshes

### CODE REVIEW

File: gui/src/3D/behaviors/defaultBehavior.ts
Comment (RaananW) [inline gui/src/3D/behaviors/defaultBehavior.ts:93]: is there a way to add/remove meshes from the meshes array?

File: gui/src/3D/controls/nearMenu.ts
Comment (RaananW) [inline gui/src/3D/controls/nearMenu.ts:156]: what is the reason behind the change here? is this something we need to store as a readonly static property?

**Review by RaananW — APPROVED** [review]
LGTM!

Just 2 questions.

File: gui/src/3D/behaviors/defaultBehavior.ts
Comment (CraigFeldspar) [reply gui/src/3D/behaviors/defaultBehavior.ts:93]: Now yes, `control.defaultBehavior.surfaceMagnetismBehavior.meshes`. It's a bit tedious, so it may be better to keep a reference of the array

File: gui/src/3D/controls/nearMenu.ts
Comment (CraigFeldspar) [reply gui/src/3D/controls/nearMenu.ts:161]: that's a mistake, I'm reverting

Comment (CraigFeldspar) [reply gui/src/3D/controls/nearMenu.ts:161]: It is changed again in a more recent PR (#10485), to be a fixed value for the near menu. I will add a parameter in this PR instead

**Comment by RaananW** [comment]
There are conflicts and a build error regarding `_defaultBehavior`. I'll merge right after it's fixes

**Pull request merged**
